tests/components/backup/test_manager.py

```diff
@@ -911,3 +919,79 @@ async def test_async_trigger_restore_wrong_parameters(
         pytest.raises(HomeAssistantError, match=expected_error),
     ):
         await manager.async_restore_backup(**(default_parameters | parameters))
+
+
+@pytest.mark.parametrize(
+    "command",
+    [
+        {
+            "type": "backup/config/update",
+        },
+        {
+            "type": "backup/config/update",
+            "create_backup": {
+                "agent_ids": ["backup.local"],
+                "include_addons": ["test-addon"],
+                "include_folders": ["media"],
+                "name": "test-name",
+                "password": "test-password",
+            },
+            "schedule": "daily",
+        },
+    ],
+)
+async def test_loading_saving_data(
+    hass: HomeAssistant,
+    hass_storage: dict[str, Any],
+    hass_ws_client: WebSocketGenerator,
+    create_backup: AsyncMock,
+    freezer: FrozenDateTimeFactory,
+    command: dict[str, Any],
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test loading and saving data."""
+    ws_client = await hass_ws_client(hass)
+
+    await hass.config.async_set_time_zone("Europe/Amsterdam")
+    freezer.move_to("2024-11-11 12:00:00+01:00")
+
+    assert await async_setup_component(hass, DOMAIN, {})
+    await hass.async_block_till_done()
+    manager1 = hass.data[DATA_MANAGER]
+
+    # Configure the manager
+    await ws_client.send_json_auto_id(command)
+    result = await ws_client.receive_json()
+    assert result["success"]
+
+    freezer.move_to("2024-11-12 12:00:00+01:00")
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    # Trigger store save
+    freezer.tick()
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    assert hass_storage[DOMAIN]["data"] == snapshot
+
+    manager2 = BackupManager(hass, CoreBackupReaderWriter(hass))
+    await manager2.async_setup()
+
+    def compare_config_dataclass(obj1: DataclassInstance, obj2: Any) -> None:
+        """Compare two BackupConfigData instances."""
+        for field in obj1.__dataclass_fields__:
+            value1 = getattr(obj1, field)
+            value2 = getattr(obj2, field)
+            if isinstance(value1, CronSim):
+                assert value1.explain() == value2.explain()
+            elif isinstance(value1, StrEnum):
+                assert type(value1) is type(value2)
+                assert value1 == value2
+            elif is_dataclass(value1):
+                compare_config_dataclass(value1, value2)
+            else:
+                assert value1 == value2
+
+    compare_config_dataclass(manager1.config.data, manager2.config.data)
```

Comment:
We shouldn't involve the manager in this unit test. This test should be in `test_config.py` and just test the serialization of the dataclass, ie the from and to dict methods. We don't need to set up the backup integration at all.

Reply:
We need a test which saves to storage and then loads the stored data back, we've had a lot of bugs caused by the lack of such tests in other cases.

What we could do is maybe:
- Adjust the test added in this PR so we don't mess directly with the manager, we instead setup the backup integration a second time and verify the scheduling still works and we can read back the config via websocket commands?
- Add separate tests in `test_config.py` which just test the serialization and deserialization

Reply:
We already have tests that tests loading and writing from/to storage. They are failing now with this fix as the tests were wrong.

But we can add additional tests per your suggestion above.

Reply:
I didn't notice we had tests which fail with the bugfix in place.

I've now fixed the existing tests and removed the new tests. I'll add new tests in a follow-up instead.